src/dotnet/APIView/APIViewWeb/Repositories/CosmosPullRequestsRepository.cs

```diff
@@ -19,9 +19,10 @@ public CosmosPullRequestsRepository(IConfiguration configuration)
             _pullRequestsContainer = client.GetContainer("APIView", "PullRequests");
         }
 
-        public async Task<PullRequestModel> GetPullRequestAsync(int pullRequestNumber, string repoName, string packageName)
+        public async Task<PullRequestModel> GetPullRequestAsync(int pullRequestNumber, string repoName,
+            string packageName, string filePath)
         {
-            var query = $"SELECT * FROM PullRequests c WHERE c.PullRequestNumber = {pullRequestNumber} and c.RepoName = '{repoName}' and c.PackageName = '{packageName}'";
+            var query = $"SELECT * FROM PullRequests c WHERE c.PullRequestNumber = {pullRequestNumber} and c.RepoName = '{repoName}' and c.PackageName = '{packageName}' and c.FilePath = '{filePath}'";
```

Comment:
Hi @praveenkuttappan  I add a more condition in querying `pullRequestModel` because I find the root cause of creating java apiview failed: if there is another language sdk has the same package name and have created apiview, java get the pull request model of the one created by another language. For example: in spec PR, python sdk usually has the same name as java sdk. When python sdk creates apiview succesfully, java sdk will find the `pullRequestModel` python sdk created.

Reply:
I am a bit worried about this change. File name should not be used since it can be empty in some scenarios once we implement swagger API review from PR. Repo name was added in the query to avoid this cross-language name conflict. What's the repo name you are passing to APIView from your caller side? Is it Spec repo name or individual language? 

Better solution is to add language name as a new param to this API and use it as additional query clause. But we need to take existing data without this property also into consideration. I can submit a change for it now. you can revert just this change for java. I will have a PR ready in few minutes.

Reply:
reverted